src/send/mod.rs

```diff
@@ -0,0 +1,99 @@
+#[cfg(test)]
+mod test;
+
+use rocket::{
+  data::{self, FromData},
+  http::Status,
+  Data,
+  Outcome,
+  Request,
+};
+use rocket_contrib::{Json, Value};
+use std::{borrow::Cow, collections::HashMap};
+use validator::{self, ValidationError};
+
+#[derive(Debug, Deserialize)]
+struct Body
+{
+  text: String,
+  html: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+struct Email
+{
+  to: Vec<String>,
+  cc: Option<Vec<String>>,
+  subject: String,
+  body: Body,
+}
+
+impl FromData for Email
+{
+  type Error = ValidationError;
+
+  fn from_data(request: &Request, data: Data) -> data::Outcome<Self, Self::Error>
+  {
+    let result = Json::<Email>::from_data(request, data);
```

Comment:
I recommend avoiding the Outcome API as much as possible. The code below and fns like validate_address will be more straightforward built around the basic Result/Option types (and maybe easier to get it looking like bbangert's suggestion below).

Defer converting to Outcome until as late as possible. I've gone as far as having the request guard call out to a separate function returning probably a Result, then doing a final conversion